dspace-api/src/main/java/org/dspace/checker/dao/impl/MostRecentChecksumDAOImpl.java

```diff
@@ -112,11 +127,18 @@ public MostRecentChecksum getOldestRecord(Context context) throws SQLException {
         //        + "from most_recent_checksum " + "where to_be_processed = true "
         //        + "order by date_trunc('milliseconds', last_process_end_date), "
         //        + "bitstream_id " + "ASC LIMIT 1";
-        Criteria criteria = createCriteria(context, MostRecentChecksum.class);
-        criteria.add(Restrictions.eq("toBeProcessed", true));
-        criteria.addOrder(Order.asc("processEndDate")).addOrder(Order.asc("bitstream.id"));
-        criteria.setMaxResults(1);
-        return singleResult(criteria);
+
+
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery<MostRecentChecksum> criteriaQuery = getCriteriaQuery(criteriaBuilder, MostRecentChecksum.class);
+        Root<MostRecentChecksum> mostRecentChecksumRoot = criteriaQuery.from(MostRecentChecksum.class);
+        criteriaQuery.select(mostRecentChecksumRoot);
+        criteriaQuery.where(criteriaBuilder.equal(mostRecentChecksumRoot.get(MostRecentChecksum_.toBeProcessed), true));
+        List<Order> orderList = new LinkedList<>();
+        orderList.add(criteriaBuilder.asc(mostRecentChecksumRoot.get(MostRecentChecksum_.bitstream)));
+        orderList.add(criteriaBuilder.asc(mostRecentChecksumRoot.get(MostRecentChecksum_.processEndDate)));
```

Comment:
Comparing to current code, it looks like these two sort orders are now added in the opposite order?  In current code it looks to be `processEndDate` *then* `bitstream`.  Is this new ordering on purpose? Or is there a mistake here?